include/dsn/c/api_layer1.h

```diff
@@ -785,6 +785,9 @@ extern DSN_API void          dsn_msg_write_next(
 /*! commit the write buffer after the message content is written with the real written size */
 extern DSN_API void          dsn_msg_write_commit(dsn_message_t msg, size_t size);
 
+/*! append a buffer to the message, the "data" can be reinterpreted cast to (blob*)  */
+extern DSN_API void          dsn_msg_write_append(dsn_message_t msg, dsn_blob_t data);
```

Comment:
The holder is a shared_ptr, so the life-time seems unable to handle.... can
you think out some resolution to avoid memory copy?

On Fri, Jul 29, 2016 at 4:44 PM, Zhenyu Guo <notifications@github.com>
wrote:

> In include/dsn/c/api_layer1.h
> <https://github.com/imzhenyu/rDSN/pull/499#discussion_r72759513>:
>
> > @@ -785,6 +785,9 @@ extern DSN_API void          dsn_msg_write_next(
> >  /*! commit the write buffer after the message content is written with the real written size */
> >  extern DSN_API void          dsn_msg_write_commit(dsn_message_t msg, size_t size);
> >
> > +/*! append a buffer to the message, the "data" can be reinterpreted cast to (blob*)  */
> > +extern DSN_API void          dsn_msg_write_append(dsn_message_t msg, dsn_blob_t data);
>
> The caller needs to handle the life-time explicitly; or we have to copy
> data.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/imzhenyu/rDSN/pull/499/files/ea230134983ab865c8bb7b6e2110776c7d14b9ac#r72759513>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAgQPXyekSSBRkBBDcDa5ne71m3UAcGTks5qab1wgaJpZM4JX4Dq>
> .
>